From aac9678287f1a35e927667d4ada48ddb840c2945 Mon Sep 17 00:00:00 2001
From: Andrew Grieve <agrieve@chromium.org>
Date: Tue, 11 May 2021 18:40:47 +0000
Subject: [PATCH] Py2->Py3 for //third_party/dom_distiller_js

Changes getting plugin_pb2.py from
  //third_party/dom_distiller_js/dist
to
  //third_party/protobuf/python/google

The dist copy was not py3 compatible.

Bug: 1205622
Change-Id: I3844674e1e0af3cbeda774be3e2fa3acb2e2640c
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2874028
Reviewed-by: Tommy Nyquist <nyquist@chromium.org>
Commit-Queue: Andrew Grieve <agrieve@chromium.org>
Cr-Commit-Position: refs/heads/master@{#881617}
---
 src/third_party/dom_distiller_js/BUILD.gn     |  15 +-
 .../protoc_plugins/json_values_converter.py   |   3 +-
 .../json_values_converter_tests.py            |   4 +-
 .../protoc_plugins/util/plugin_protos.py      |  11 +-
 .../update_domdistiller_js.sh                 |   4 -
 src/third_party/protobuf/.gitignore           |   1 +
 src/third_party/protobuf/README.chromium      |   4 +-
 .../protobuf/gen_extra_chromium_files         |  22 --
 .../protobuf/gen_extra_chromium_files.py      |  44 +++
 .../google/protobuf/compiler/plugin_pb2.py    | 301 ++++++++++++++++++
 10 files changed, 365 insertions(+), 44 deletions(-)
 delete mode 100755 src/third_party/protobuf/gen_extra_chromium_files
 create mode 100755 src/third_party/protobuf/gen_extra_chromium_files.py
 create mode 100644 src/third_party/protobuf/python/google/protobuf/compiler/plugin_pb2.py

diff --git a/src/third_party/dom_distiller_js/BUILD.gn b/src/third_party/dom_distiller_js/BUILD.gn
index b8d1918342..7096af74b8 100644
--- a/src/third_party/dom_distiller_js/BUILD.gn
+++ b/src/third_party/dom_distiller_js/BUILD.gn
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//build/config/python.gni")
 import("//third_party/protobuf/proto_library.gni")
 
 group("proto") {
@@ -51,18 +50,22 @@ proto_library("json_values_converter_test_proto") {
   generator_plugin_options = "output_dir=:"
 }
 
-# TODO(crbug.com/1112471): Get this to run cleanly under Python 3.
-python2_action("json_values_converter_tests") {
+action("json_values_converter_tests") {
   _stamp = "$target_gen_dir/json_values_converter_tests.stamp"
+  _actual_file = "$target_gen_dir/test_sample_json_converter.h"
+  _golden_file = "test_sample_json_converter.h.golden"
   deps = [ ":json_values_converter_test_proto" ]
-  inputs = [ "test_sample_json_converter.h.golden" ]
+  inputs = [
+    _actual_file,
+    _golden_file,
+  ]
   outputs = [ _stamp ]
 
   script = "protoc_plugins/json_values_converter_tests.py"
   args = [
     "--stamp",
     rebase_path(_stamp, root_build_dir),
-    rebase_path("$target_gen_dir/test_sample_json_converter.h", root_build_dir),
-    rebase_path("test_sample_json_converter.h.golden", root_build_dir),
+    rebase_path(_actual_file, root_build_dir),
+    rebase_path(_golden_file, root_build_dir),
   ]
 }
diff --git a/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py b/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
index e86a88c759..954a410275 100755
--- a/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
+++ b/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
@@ -235,7 +235,8 @@ def SetBinaryStdio():
 
 
 def ReadRequestFromStdin():
-  data = sys.stdin.read()
+  stream = sys.stdin if sys.version_info[0] < 3 else sys.stdin.buffer
+  data = stream.read()
   return plugin_protos.PluginRequestFromString(data)
 
 
diff --git a/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter_tests.py b/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter_tests.py
index 53d428cc3d..031898388d 100755
--- a/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter_tests.py
+++ b/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter_tests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright (c) 2016 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
@@ -14,7 +14,7 @@ import sys
 
 
 def CompareFiles(file1, file2):
-  return file(file1, 'r').read() == file(file2, 'r').read()
+  return open(file1, 'r').read() == open(file2, 'r').read()
 
 
 def TouchStamp(stamp_path):
diff --git a/src/third_party/dom_distiller_js/protoc_plugins/util/plugin_protos.py b/src/third_party/dom_distiller_js/protoc_plugins/util/plugin_protos.py
index 5bbfd83a91..d3cbaf1a87 100644
--- a/src/third_party/dom_distiller_js/protoc_plugins/util/plugin_protos.py
+++ b/src/third_party/dom_distiller_js/protoc_plugins/util/plugin_protos.py
@@ -20,14 +20,10 @@ sys.path.insert(
 sys.path.insert(
     1, os.path.join(SRC_DIR, 'third_party', 'protobuf', 'third_party', 'six'))
 from google.protobuf.descriptor_pb2 import FieldDescriptorProto
+from google.protobuf.compiler import plugin_pb2
 
 from . import types
 
-sys.path.insert(
-    1, os.path.join(SRC_DIR,
-                    'third_party', 'dom_distiller_js', 'dist', 'python'))
-import plugin_pb2
-
 
 class PluginRequest(object):
   def __init__(self, proto):
@@ -62,8 +58,9 @@ class PluginResponse(object):
     self.proto.error += err + '\n'
 
   def WriteToStdout(self):
-    sys.stdout.write(self.proto.SerializeToString())
-    sys.stdout.flush()
+    stream = sys.stdout if sys.version_info[0] < 3 else sys.stdout.buffer
+    stream.write(self.proto.SerializeToString())
+    stream.flush()
 
 
 class ProtoFile(object):
diff --git a/src/third_party/dom_distiller_js/update_domdistiller_js.sh b/src/third_party/dom_distiller_js/update_domdistiller_js.sh
index e3f7d707db..4fcf3adae4 100755
--- a/src/third_party/dom_distiller_js/update_domdistiller_js.sh
+++ b/src/third_party/dom_distiller_js/update_domdistiller_js.sh
@@ -76,10 +76,6 @@
 
   cp -r $tmpdir/dom-distiller/out/package/* .
 
-  # Stop rolling python/plugin_pb2.py for protobuf backward compatibility.
-  # See https://crbug.com/874509
-  git checkout -- python/plugin_pb2.py
-
   git add .
   if [[ $(git status --short | wc -l) -ne 0 ]]; then
     if [[ -n "$gerrit_url" ]]; then
diff --git a/src/third_party/protobuf/.gitignore b/src/third_party/protobuf/.gitignore
index 291fa8f781..9a935e339f 100644
--- a/src/third_party/protobuf/.gitignore
+++ b/src/third_party/protobuf/.gitignore
@@ -54,6 +54,7 @@ src/google/protobuf/util/**/*.pb.h
 *.egg-info
 *_pb2.py
 !python/google/protobuf/descriptor_pb2.py
+!python/google/protobuf/compiler/plugin_pb2.py
 python/**/*.egg
 python/.eggs/
 python/.tox
diff --git a/src/third_party/protobuf/README.chromium b/src/third_party/protobuf/README.chromium
index ec90f30eeb..f27b0848d7 100644
--- a/src/third_party/protobuf/README.chromium
+++ b/src/third_party/protobuf/README.chromium
@@ -31,8 +31,8 @@ Steps used to create the current version:
    For future releases, it will be worth looking into which patches still need
    to be applied. In case of conflict, update those patches accordingly and save
    them back in place (i.e. in patches directory).
-8. Generate descriptor_pb2.py using the script "gen_extra_chromium_files" in
-the same directory as this file.
+8. Generate descriptor_pb2.py using the script "gen_extra_chromium_files.py" in
+   the same directory as this file.
 
 9. Add an __init__.py to protobuf/ that adds third_party/six/ to Python path.
 10. Update this file (README.chromium).
diff --git a/src/third_party/protobuf/gen_extra_chromium_files b/src/third_party/protobuf/gen_extra_chromium_files
deleted file mode 100755
index a32f2b942e..0000000000
--- a/src/third_party/protobuf/gen_extra_chromium_files
+++ /dev/null
@@ -1,22 +0,0 @@
-#!/usr/bin/env python
-
-# This script expects your out/Debug directory to be compiling for linux on
-# a linux machine. If this is not your case just compile protoc and run the
-# command on the last line of the script (from within
-# //third_party/protobuf/src).
-
-import os
-src_dir = os.path.abspath(os.path.join(os.path.dirname(__file__),'..','..'))
-out_build_dir = 'out/Debug'
-
-def r(c):
-  print c
-  os.system(c)
-
-os.chdir(src_dir)
-r('gn gen {}'.format(out_build_dir))
-r('ninja -C {} protoc'.format(out_build_dir))
-os.chdir('third_party/protobuf/src')
-protoc = os.path.join(src_dir,'{}/protoc'.format(out_build_dir))
-r('{} --python_out=../python google/protobuf/descriptor.proto'.format(protoc))
-
diff --git a/src/third_party/protobuf/gen_extra_chromium_files.py b/src/third_party/protobuf/gen_extra_chromium_files.py
new file mode 100755
index 0000000000..a109e86090
--- /dev/null
+++ b/src/third_party/protobuf/gen_extra_chromium_files.py
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+
+# This script expects your out/Debug directory to be compiling for linux on
+# a linux machine. If this is not your case just compile protoc and run the
+# command on the last line of the script (from within
+# //third_party/protobuf/src).
+
+import argparse
+import os
+
+PROTO_DIR = os.path.dirname(__file__)
+DIR_SOURCE_ROOT = os.path.join(PROTO_DIR, '..','..')
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--output-directory',
+                        '-C',
+                        help='Path to build directory',
+                        required=True)
+
+    args = parser.parse_args()
+
+    out_build_dir = args.output_directory
+
+    def r(c):
+        print('Running:', c)
+        os.system(c)
+
+    r('ninja -C {} protoc'.format(out_build_dir))
+
+    protoc = os.path.join(out_build_dir, 'protoc')
+    print('Creating //third_party/protobuf/python/google/protobuf/'
+          'descriptor_pb2.py')
+    r('{0} --proto_path={1}/src --python_out={1}/python '
+      '{1}/src/google/protobuf/descriptor.proto'.format(protoc, PROTO_DIR))
+
+    print('Creating //third_party/protobuf/python/google/protobuf/compiler/'
+          'plugin_pb2.py')
+    r('{0} --proto_path={1}/src --python_out={1}/python '
+      '{1}/src/google/protobuf/compiler/plugin.proto'.format(protoc, PROTO_DIR))
+
+if __name__ == '__main__':
+    main()
diff --git a/src/third_party/protobuf/python/google/protobuf/compiler/plugin_pb2.py b/src/third_party/protobuf/python/google/protobuf/compiler/plugin_pb2.py
new file mode 100644
index 0000000000..ff90d18f1b
--- /dev/null
+++ b/src/third_party/protobuf/python/google/protobuf/compiler/plugin_pb2.py
@@ -0,0 +1,301 @@
+# -*- coding: utf-8 -*-
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
+# source: google/protobuf/compiler/plugin.proto
+"""Generated protocol buffer code."""
+from google.protobuf import descriptor as _descriptor
+from google.protobuf import message as _message
+from google.protobuf import reflection as _reflection
+from google.protobuf import symbol_database as _symbol_database
+# @@protoc_insertion_point(imports)
+
+_sym_db = _symbol_database.Default()
+
+
+from google.protobuf import descriptor_pb2 as google_dot_protobuf_dot_descriptor__pb2
+
+
+DESCRIPTOR = _descriptor.FileDescriptor(
+  name='google/protobuf/compiler/plugin.proto',
+  package='google.protobuf.compiler',
+  syntax='proto2',
+  serialized_options=b'\n\034com.google.protobuf.compilerB\014PluginProtosZ)google.golang.org/protobuf/types/pluginpb',
+  create_key=_descriptor._internal_create_key,
+  serialized_pb=b'\n%google/protobuf/compiler/plugin.proto\x12\x18google.protobuf.compiler\x1a google/protobuf/descriptor.proto\"F\n\x07Version\x12\r\n\x05major\x18\x01 \x01(\x05\x12\r\n\x05minor\x18\x02 \x01(\x05\x12\r\n\x05patch\x18\x03 \x01(\x05\x12\x0e\n\x06suffix\x18\x04 \x01(\t\"\xba\x01\n\x14\x43odeGeneratorRequest\x12\x18\n\x10\x66ile_to_generate\x18\x01 \x03(\t\x12\x11\n\tparameter\x18\x02 \x01(\t\x12\x38\n\nproto_file\x18\x0f \x03(\x0b\x32$.google.protobuf.FileDescriptorProto\x12;\n\x10\x63ompiler_version\x18\x03 \x01(\x0b\x32!.google.protobuf.compiler.Version\"\xc1\x02\n\x15\x43odeGeneratorResponse\x12\r\n\x05\x65rror\x18\x01 \x01(\t\x12\x1a\n\x12supported_features\x18\x02 \x01(\x04\x12\x42\n\x04\x66ile\x18\x0f \x03(\x0b\x32\x34.google.protobuf.compiler.CodeGeneratorResponse.File\x1a\x7f\n\x04\x46ile\x12\x0c\n\x04name\x18\x01 \x01(\t\x12\x17\n\x0finsertion_point\x18\x02 \x01(\t\x12\x0f\n\x07\x63ontent\x18\x0f \x01(\t\x12?\n\x13generated_code_info\x18\x10 \x01(\x0b\x32\".google.protobuf.GeneratedCodeInfo\"8\n\x07\x46\x65\x61ture\x12\x10\n\x0c\x46\x45\x41TURE_NONE\x10\x00\x12\x1b\n\x17\x46\x45\x41TURE_PROTO3_OPTIONAL\x10\x01\x42W\n\x1c\x63om.google.protobuf.compilerB\x0cPluginProtosZ)google.golang.org/protobuf/types/pluginpb'
+  ,
+  dependencies=[google_dot_protobuf_dot_descriptor__pb2.DESCRIPTOR,])
+
+
+
+_CODEGENERATORRESPONSE_FEATURE = _descriptor.EnumDescriptor(
+  name='Feature',
+  full_name='google.protobuf.compiler.CodeGeneratorResponse.Feature',
+  filename=None,
+  file=DESCRIPTOR,
+  create_key=_descriptor._internal_create_key,
+  values=[
+    _descriptor.EnumValueDescriptor(
+      name='FEATURE_NONE', index=0, number=0,
+      serialized_options=None,
+      type=None,
+      create_key=_descriptor._internal_create_key),
+    _descriptor.EnumValueDescriptor(
+      name='FEATURE_PROTO3_OPTIONAL', index=1, number=1,
+      serialized_options=None,
+      type=None,
+      create_key=_descriptor._internal_create_key),
+  ],
+  containing_type=None,
+  serialized_options=None,
+  serialized_start=628,
+  serialized_end=684,
+)
+_sym_db.RegisterEnumDescriptor(_CODEGENERATORRESPONSE_FEATURE)
+
+
+_VERSION = _descriptor.Descriptor(
+  name='Version',
+  full_name='google.protobuf.compiler.Version',
+  filename=None,
+  file=DESCRIPTOR,
+  containing_type=None,
+  create_key=_descriptor._internal_create_key,
+  fields=[
+    _descriptor.FieldDescriptor(
+      name='major', full_name='google.protobuf.compiler.Version.major', index=0,
+      number=1, type=5, cpp_type=1, label=1,
+      has_default_value=False, default_value=0,
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='minor', full_name='google.protobuf.compiler.Version.minor', index=1,
+      number=2, type=5, cpp_type=1, label=1,
+      has_default_value=False, default_value=0,
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='patch', full_name='google.protobuf.compiler.Version.patch', index=2,
+      number=3, type=5, cpp_type=1, label=1,
+      has_default_value=False, default_value=0,
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='suffix', full_name='google.protobuf.compiler.Version.suffix', index=3,
+      number=4, type=9, cpp_type=9, label=1,
+      has_default_value=False, default_value=b"".decode('utf-8'),
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+  ],
+  extensions=[
+  ],
+  nested_types=[],
+  enum_types=[
+  ],
+  serialized_options=None,
+  is_extendable=False,
+  syntax='proto2',
+  extension_ranges=[],
+  oneofs=[
+  ],
+  serialized_start=101,
+  serialized_end=171,
+)
+
+
+_CODEGENERATORREQUEST = _descriptor.Descriptor(
+  name='CodeGeneratorRequest',
+  full_name='google.protobuf.compiler.CodeGeneratorRequest',
+  filename=None,
+  file=DESCRIPTOR,
+  containing_type=None,
+  create_key=_descriptor._internal_create_key,
+  fields=[
+    _descriptor.FieldDescriptor(
+      name='file_to_generate', full_name='google.protobuf.compiler.CodeGeneratorRequest.file_to_generate', index=0,
+      number=1, type=9, cpp_type=9, label=3,
+      has_default_value=False, default_value=[],
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='parameter', full_name='google.protobuf.compiler.CodeGeneratorRequest.parameter', index=1,
+      number=2, type=9, cpp_type=9, label=1,
+      has_default_value=False, default_value=b"".decode('utf-8'),
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='proto_file', full_name='google.protobuf.compiler.CodeGeneratorRequest.proto_file', index=2,
+      number=15, type=11, cpp_type=10, label=3,
+      has_default_value=False, default_value=[],
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='compiler_version', full_name='google.protobuf.compiler.CodeGeneratorRequest.compiler_version', index=3,
+      number=3, type=11, cpp_type=10, label=1,
+      has_default_value=False, default_value=None,
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+  ],
+  extensions=[
+  ],
+  nested_types=[],
+  enum_types=[
+  ],
+  serialized_options=None,
+  is_extendable=False,
+  syntax='proto2',
+  extension_ranges=[],
+  oneofs=[
+  ],
+  serialized_start=174,
+  serialized_end=360,
+)
+
+
+_CODEGENERATORRESPONSE_FILE = _descriptor.Descriptor(
+  name='File',
+  full_name='google.protobuf.compiler.CodeGeneratorResponse.File',
+  filename=None,
+  file=DESCRIPTOR,
+  containing_type=None,
+  create_key=_descriptor._internal_create_key,
+  fields=[
+    _descriptor.FieldDescriptor(
+      name='name', full_name='google.protobuf.compiler.CodeGeneratorResponse.File.name', index=0,
+      number=1, type=9, cpp_type=9, label=1,
+      has_default_value=False, default_value=b"".decode('utf-8'),
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='insertion_point', full_name='google.protobuf.compiler.CodeGeneratorResponse.File.insertion_point', index=1,
+      number=2, type=9, cpp_type=9, label=1,
+      has_default_value=False, default_value=b"".decode('utf-8'),
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='content', full_name='google.protobuf.compiler.CodeGeneratorResponse.File.content', index=2,
+      number=15, type=9, cpp_type=9, label=1,
+      has_default_value=False, default_value=b"".decode('utf-8'),
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='generated_code_info', full_name='google.protobuf.compiler.CodeGeneratorResponse.File.generated_code_info', index=3,
+      number=16, type=11, cpp_type=10, label=1,
+      has_default_value=False, default_value=None,
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+  ],
+  extensions=[
+  ],
+  nested_types=[],
+  enum_types=[
+  ],
+  serialized_options=None,
+  is_extendable=False,
+  syntax='proto2',
+  extension_ranges=[],
+  oneofs=[
+  ],
+  serialized_start=499,
+  serialized_end=626,
+)
+
+_CODEGENERATORRESPONSE = _descriptor.Descriptor(
+  name='CodeGeneratorResponse',
+  full_name='google.protobuf.compiler.CodeGeneratorResponse',
+  filename=None,
+  file=DESCRIPTOR,
+  containing_type=None,
+  create_key=_descriptor._internal_create_key,
+  fields=[
+    _descriptor.FieldDescriptor(
+      name='error', full_name='google.protobuf.compiler.CodeGeneratorResponse.error', index=0,
+      number=1, type=9, cpp_type=9, label=1,
+      has_default_value=False, default_value=b"".decode('utf-8'),
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='supported_features', full_name='google.protobuf.compiler.CodeGeneratorResponse.supported_features', index=1,
+      number=2, type=4, cpp_type=4, label=1,
+      has_default_value=False, default_value=0,
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+    _descriptor.FieldDescriptor(
+      name='file', full_name='google.protobuf.compiler.CodeGeneratorResponse.file', index=2,
+      number=15, type=11, cpp_type=10, label=3,
+      has_default_value=False, default_value=[],
+      message_type=None, enum_type=None, containing_type=None,
+      is_extension=False, extension_scope=None,
+      serialized_options=None, file=DESCRIPTOR,  create_key=_descriptor._internal_create_key),
+  ],
+  extensions=[
+  ],
+  nested_types=[_CODEGENERATORRESPONSE_FILE, ],
+  enum_types=[
+    _CODEGENERATORRESPONSE_FEATURE,
+  ],
+  serialized_options=None,
+  is_extendable=False,
+  syntax='proto2',
+  extension_ranges=[],
+  oneofs=[
+  ],
+  serialized_start=363,
+  serialized_end=684,
+)
+
+_CODEGENERATORREQUEST.fields_by_name['proto_file'].message_type = google_dot_protobuf_dot_descriptor__pb2._FILEDESCRIPTORPROTO
+_CODEGENERATORREQUEST.fields_by_name['compiler_version'].message_type = _VERSION
+_CODEGENERATORRESPONSE_FILE.fields_by_name['generated_code_info'].message_type = google_dot_protobuf_dot_descriptor__pb2._GENERATEDCODEINFO
+_CODEGENERATORRESPONSE_FILE.containing_type = _CODEGENERATORRESPONSE
+_CODEGENERATORRESPONSE.fields_by_name['file'].message_type = _CODEGENERATORRESPONSE_FILE
+_CODEGENERATORRESPONSE_FEATURE.containing_type = _CODEGENERATORRESPONSE
+DESCRIPTOR.message_types_by_name['Version'] = _VERSION
+DESCRIPTOR.message_types_by_name['CodeGeneratorRequest'] = _CODEGENERATORREQUEST
+DESCRIPTOR.message_types_by_name['CodeGeneratorResponse'] = _CODEGENERATORRESPONSE
+_sym_db.RegisterFileDescriptor(DESCRIPTOR)
+
+Version = _reflection.GeneratedProtocolMessageType('Version', (_message.Message,), {
+  'DESCRIPTOR' : _VERSION,
+  '__module__' : 'google.protobuf.compiler.plugin_pb2'
+  # @@protoc_insertion_point(class_scope:google.protobuf.compiler.Version)
+  })
+_sym_db.RegisterMessage(Version)
+
+CodeGeneratorRequest = _reflection.GeneratedProtocolMessageType('CodeGeneratorRequest', (_message.Message,), {
+  'DESCRIPTOR' : _CODEGENERATORREQUEST,
+  '__module__' : 'google.protobuf.compiler.plugin_pb2'
+  # @@protoc_insertion_point(class_scope:google.protobuf.compiler.CodeGeneratorRequest)
+  })
+_sym_db.RegisterMessage(CodeGeneratorRequest)
+
+CodeGeneratorResponse = _reflection.GeneratedProtocolMessageType('CodeGeneratorResponse', (_message.Message,), {
+
+  'File' : _reflection.GeneratedProtocolMessageType('File', (_message.Message,), {
+    'DESCRIPTOR' : _CODEGENERATORRESPONSE_FILE,
+    '__module__' : 'google.protobuf.compiler.plugin_pb2'
+    # @@protoc_insertion_point(class_scope:google.protobuf.compiler.CodeGeneratorResponse.File)
+    })
+  ,
+  'DESCRIPTOR' : _CODEGENERATORRESPONSE,
+  '__module__' : 'google.protobuf.compiler.plugin_pb2'
+  # @@protoc_insertion_point(class_scope:google.protobuf.compiler.CodeGeneratorResponse)
+  })
+_sym_db.RegisterMessage(CodeGeneratorResponse)
+_sym_db.RegisterMessage(CodeGeneratorResponse.File)
+
+
+DESCRIPTOR._options = None
+# @@protoc_insertion_point(module_scope)
